### PR TITLE
Fixed timezone issue in dom.Timestamp

### PR DESCRIPTION
**Pull request by zslayton**
*Issue #, if available:* #595 

*Description of changes:*

* When creating a `dom.Timestamp` from a provided `Date`, stores the UTC datetime represented by the `Date`. (See issue #595 for an explanation of why storing the `Date`'s offset is not meaningful.)
* Modifies an existing unit test to catch this sort of error in the event of a regression.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by zslayton** [comment]
>  can we still assert `domValue.dateValue()` behavior is correct?

I removed this check because `Value.from(myDate).dateValue()` is guaranteed to be `myDate`. However, I only know this is true because I wrote the implementation; it certainly doesn't hurt to test it. I've re-added the check.
 
> the new comment suggests passing in an ion.Timestamp -- do we have a test for this?

Yes, in a few places. For example, [these round-trip tests](https://github.com/amzn/ion-js/blob/42ce6a42a9293605f194049ee8b12146f822b89f/test/dom/Value.ts#L361-L363) create `dom.Timestamp`s from `ion.Timestamps`.

> we have a test for offset Z, but I'd suggest having a test for a non-zero offset

I've added a bunch of example `Date` tests to check for this and other corner cases. In the process, I fixed an issue that could crop with fractional seconds (which, it turns out, `ion.Timestamp` requires be specified as an `ion.Decimal`). 👍

**Comment by pbcornell** [comment]
> In the process, I fixed an issue that could crop with fractional seconds (which, it turns out, `ion.Timestamp` requires be specified as an `ion.Decimal`). 👍

Ok great...but now I'm looking for tests that assert expected behavior for non-zero milliseconds (for Date), nor any tests for micro-/nano-seconds (when using Timestamp) and I don't see any.

**Comment by zslayton** [comment]
Changes:
* I refactored `exampleValues.ts` so it now has a list of ISO-8601 strings that is used to provide arrays of both sample `Date`s and `Timestamp`s.
* I added lots of different fractional-second ISO-8601 examples.
* I changed some existing `Timestamp` unit tests to use the new `exampleTimestampsWhere()` values, which should be much more comprehensive.

File: test/exampleValues.ts [hunk 1]
```diff
@@ -41,4 +41,49 @@ export function exampleIonValuesWhere(filter: (v: Value) => boolean = acceptAnyV
...
+    "2020-03-28T01:00:00.123456+01:00",
+    "2020-03-28T01:00:00.123456789+01:00",
+];
+
+// A common collection of Date values that can be reduced to a relevant subset using
+// the provided filter function.
+export function exampleDatesWhere(filter: (v: Date) => boolean = acceptAnyValue): Date[] {
+    return _exampleIsoStrings
+    .map((isoString) => new Date(isoString))
+    .filter(filter);
```
Comment (pbcornell) [inline test/exampleValues.ts]: indent
